### ISSUE TITLE
Agents stopped taking work (getting windows license error on agent)

### ISSUE THREAD

**Issue by rdejuana** [WPO-Foundation/webpagetest#1219]
I am running an AWS private instance and the agents stopped taking work.  I have rebooted a few times and everything seems to be ok on the master.  We took a screen shot saw a windows license error.

![webpagetest-windows](https://user-images.githubusercontent.com/23434/48359544-cde01b80-e662-11e8-9090-e98af8836479.jpeg)

I don't know if this is normal, but this is the only error we have seen.

**Comment by pmeenan** [comment]
Never seen the licence error before.  The RDP warning was also just reported here: https://github.com/WPO-Foundation/wptagent/issues/213

I'll see about updating the Windows AMI's and installing all of the updates to see if that cleans it up.

**Comment by pmeenan** [comment]
The AMIs have all [been updated](https://github.com/WPO-Foundation/webpagetest/commit/5e207ff6195b1e2388dbc9da8af59d68c195662d) to fix the RDP error (and pick up the last 5 months of Windows updates).

I didn't see anything about the license or activation issues on mine but the core issue was likely the RDP issue.

Servers should pick up the change within the next hour (or you can nudge it by hitting http://your.server.name/cron/hourly.php).

**Comment by pmeenan** [comment]
The network dialog shouldn't impact anything. Can you dismiss it, scroll the embedded RDP window to the bottom and open the cmd console that is running inside of the RDP window to see what it is doing?

**Comment by bentzio** [comment]
I am having the same issue. The other day our test agent stopped processing tests, and when we logged in via RDP, we got the windows license alert. That was possibly due to the widespread licensing issue Microsoft had last week. That's gone now, but I did get the same Networks alert. I went ahead and opened the embeded RDP CMD app and got a screen grab of what it was doing. I blocked out the interfaces portion as it contains ip addresses that I'm not really wanting to share with the world. Hope this helps.
![capture1](https://user-images.githubusercontent.com/20196436/48463771-d3dd1600-e7a2-11e8-9aa7-ba87feedaf74.PNG)

**Comment by pmeenan** [comment]
Those are the requests to the instance metadata server to get the user data that contains the configuration information. There should be [some code](https://github.com/WPO-Foundation/wptagent/blob/master/internal/webpagetest.py#L173) to reset the routes to help with that. Under the blocked out content anywhere is there a call out to powershell to run support/win_routes.ps1?

**Comment by bentzio** [comment]
Sory, no. I don't see anything pertaining to **support/win_routes.ps1** - I went ahead and unblocked the section and just removed the actuall addresses:

![capture1](https://user-images.githubusercontent.com/20196436/48489454-ba5ebd00-e7e8-11e8-8015-5a948828702a.PNG)

**Comment by thedoc31** [comment]
I saw the same thing on a Server2012R2 node earlier this week, but rebooting the agent seemed to resolve the issue for me.

**Comment by rdejuana** [comment]
rebooting is not fixing it for me.

**Comment by rdejuana** [comment]
FYI, we converted the agents to run linux instead of windows.  We were testing on chrome, so not having IE or safari is not a big deal for us

**Comment by bentzio** [comment]
rdejuana - the linux agent (ami) is working for you?

**Comment by pmeenan** [comment]
I'm taking another run at the Windows agents now. Looks like the failed activation is a [symptom of the routes being broken also](https://aws.amazon.com/premiumsupport/knowledge-center/windows-activation-fails/). I'm going to add a call to the AWS instance initialization script at boot before starting the RDP session and agent logic to see if that helps.

If you don't need IE, I can't recommend the Linux instances highly enough. They are used for all of the EC2 locations on the public instance, by SpeedCurve and by most everyone else I know running a private instance.  They're 1/3 the cost of the Windows instances to run and generally a LOT more reliable (and since they are used by so many production systems any issues tend to get noticed and fixed much faster).

**Comment by rdejuana** [comment]
They seem to be working fine.  And given Patrick's endorsement, it will probably stay on linux. Adding IE as an agent is a product question that I don't we're gonna tackle now.

**Comment by pmeenan** [comment]
Just pushed new Windows AMIs that run the EC2 init script at startup.  That should fix any activation or metadata issues.  It also makes it so the correct password is retrievable directly from the EC2 console (I tweaked the init script to not reset the password which would have broken the agents).

**Comment by bentzio** [comment]
So I tried switching over to using the linux ami (ami-a88c20d5), and the test agent now connects... /install shows everything required is green... but when running the tests (on both chrome and firefox), all I get back is "The test completed but there were no successful results." The documentation only touches on this being a Windows client issue, and search results indicate this doesn't happen with the AMI. I can curl the website I'm trying to test from the agent (via ssh connection). Anybody else run into this or have any ideas? Thanks so much.

**Comment by bentzio** [comment]
Odd.. I refreshed the results page and they just appeared.

**Comment by bentzio** [comment]
It's 18.10. Just pulled the linux ami (ami-a88c20d5) this morning. It seems to still be running fine at the moment. Haven't seen another "no successful results" message since the first couple.

**Comment by tkadlec** [comment]
We're in the process of cleaning up issues on this project in order to ensure we're able to stay on top of high priority bugs and feature requests. As a part of this process, we're automatically closing any issues that have had no activity within the last two years, including this one, since the codebase has changed dramatically in that time. If you feel this is still relevant, please [file a new issue](https://github.com/WPO-Foundation/webpagetest/issues/new/choose) using the relevant issue template so we can get it prioritized. Thanks!
